packages/hooks/src/useDtraderV2Flag.ts

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react';
+import useIsGrowthbookIsLoaded from './useIsGrowthbookLoaded';
+import { isDTraderV2, routes } from '@deriv/shared';
+import { useDevice } from '@deriv-com/ui';
+import { Analytics } from '@deriv-com/analytics';
+
+const useDtraderV2Flag = () => {
+    const { isGBLoaded: is_growthbook_loaded, isGBNotAvailable: is_gb_not_available } = useIsGrowthbookIsLoaded();
+    const [load_dtrader_module, setLoadDTraderModule] = useState(false);
+    const [dtrader_v2_enabled, setDTraderV2Enabled] = useState(false);
+    const { isMobile } = useDevice();
+    useEffect(() => {
+        if (is_growthbook_loaded || isDTraderV2()) {
+            setLoadDTraderModule(true);
+            setDTraderV2Enabled(
+                (isDTraderV2() || Boolean(Analytics?.getFeatureValue('dtrader_v2_enabled', false))) &&
+                    isMobile &&
+                    (location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/'))
+            );
```

Comment:
suggestion to refactor those complex if statemetns, for example like below:
```
const is_dtrader_v2 = isDTraderV2();
const is_feature_flag_active = Boolean(Analytics?.getFeatureValue('dtrader_v2_enabled', false));
const is_trade_or_contract_path = location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/');
const is_mobile = isMobile;
const should_load_dtrader_module = is_growthbook_loaded || is_dtrader_v2;

if (should_load_dtrader_module) {
    setLoadDTraderModule(true);
    const should_enable_dtrader_v2 = (is_dtrader_v2 || is_feature_flag_active) && is_mobile && is_trade_or_contract_path;
    setDTraderV2Enabled(should_enable_dtrader_v2);
}
```

complex if stataments are complexity points,

Reply:
yes made this change